src/syrupy/extensions/amber.py

```diff
@@ -242,16 +289,21 @@ def serialize(
         data: "SerializableData",
         *,
         depth: int = 0,
+        matcher: Optional["PropertyMatcher"] = None,
+        path: "PropertyPath" = (),
         visited: Optional[Set[Any]] = None,
     ) -> str:
         visited = visited if visited is not None else set()
         data_id = id(data)
         if depth > cls._max_depth or data_id in visited:
             data = cls.MarkerDepthMax()
-
+        if matcher:
+            data = matcher(data, path)
```

Comment:
I used the term "matcher" because it behaves almost exactly as jest's [asymmetric matcher](https://jestjs.io/docs/en/snapshot-testing#property-matchers)

Reply:
It looks like it behaves a bit differently. In this case, the "matcher" is a transformation function which takes some data and a path and returns a value to serialize in place of the original data. Something closer would be:

```python
def test_case(snapshot):
    assert { "a": { "b": 1, "c": "uuid" } } == snapshot(matcher=(("a","b"),))
```

```python
if matcher:
   if any(path.startswith(prefix) for prefix in matcher):
       data = Excluded(data) # special helper class for non-deterministic data
```

Reply:
I don't think it behaves in the exclusion manner, since the jest snapshots are being replaced with values like `Any<Number>` and `Any<Date>`. The difference being with jest you have limited control over what the replacement snapshot looks like.

Reply:
@noahnu see example snapshot showing usage
https://github.com/tophat/syrupy/blob/06ff63786ae63ba40d391b7e076d76a4044fc1cc/tests/__snapshots__/test_extension_amber.ambr#L217-L227